docs/guides/real-time/getting-started/websockets.md

```diff
@@ -0,0 +1,227 @@
+---
+contributors: Kevin Lewis
+description: "Learn how to get started with Directus' WebSockets interface."
+---
+
+# Getting Started With WebSockets
+
+You can connect to a Directus project using a WebSocket interface and get updates on data held in a collection in real-time.
+
+This guide will show you how to get started with Directus' WebSockets interface and JavaScript. WebSockets are language-agnostic, so you can apply the same set of steps in your stack of choice.
+
+## Before You Begin
+
+You will need a Directus project. If you don’t already have one, the easiest way to get started is with our [managed Directus Cloud service](https://directus.cloud). You can also self-host Directus, ensuring the `WEBSOCKETS_ENABLED` environment variable is set to `true`.
+
+Create a new collection called `messages`, with a `date_created` field enabled in the *Optional System Fields* pane on collection creation. Create a text field called `text` and a second called `user`.
+
+If it doesn’t already exist, create a user with a role that can execute read and create operations on the collection.
+
+Finally in the Directus Data Studio, create a static access token for the user, copy it, and save the user profile. 
+
+Create an `index.html` file and open it in your code editor. Add the following boilerplate code:
+
+```html
+<!DOCTYPE html>
+<html>
+  <body>
+    <script>
+      const url = 'wss://your-directus-url/websocket'
+      const access_token = 'your-access-token'
+      const collection = 'messages'
+    </script>
+  </body>
+</html>
+```
+
+Make sure to replace `your-directus-url` and `your-access-token` with your project and user details. 
+
+## Create a Connection
+At the bottom of your `<script>`, add the following code to establish a new WebSocket connection:
+
+```js
+const connection = new WebSocket(url)
+```
+
+To add some feedback, add the following event handlers below your `connection` variable:
+
+```js
+connection.onopen = function() {
+  console.log({ event: 'onopen' })
+}
```

Comment:
Personally disagree for a couple of reasons - conciseness of the overwrite, and this is how a lot of websockets docs write the event handler, so I mimic'd it. Not a hill I'll die on if you feel strongly - I know people can get themselves in a pickle with an overwrite.

Reply:
You can have multiple event listeners on the same object. If you use the method overwrite syntax, you will destroy your other handler. For things like the `message` event, it's not unthinkable you'd add more than 1 event listener on the same connection across multiple methods (as a matter of fact, that's most likely how our SDK will work internally as well)

Reply:
Also ease of understanding > conciseness any day. In this example, the addEventListener route makes it immediately obvious that `open` is an event you can listen to, whereas the `onopen` could be a proprietary arbitrary method on the connection object, wherever that may come from

Reply:
Fair. I did this because it matches other WebSockets guides, but I'm fine enough with helping people not shoot themselves in the foot. I'll adjust this.